rpc/ws/client.go

```diff
@@ -315,6 +379,35 @@ func (c *Client) subscribe(
 	return sub, nil
 }
 
+func (c *Client) resubscribe(requestIDs map[uint64]struct{}) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	var subscribedRequestIDs []uint64
+	for reqID := range requestIDs {
+		sub := c.subscriptionByRequestID[reqID]
+		data, err := sub.req.encode()
+		if err != nil {
+			return fmt.Errorf("subscribe: unable to encode subsciption request: %w", err)
+		}
+
+		delete(c.subscriptionByWSSubID, sub.subID)
+		sub.subID = 0 // reset subID, it will be updated later in handleNewSubscriptionMessage
```

Comment:
why are we deleting and zero out the sub's ID ?

Reply:
I zeroed out the sub id because the current sub id is outdated and will be changed later in the `handleNewSubsriptionMessage()`. Zeroing sub id helps avoid duplicated sub id.